### PR TITLE
Editing dlgDeleteSheets and  creating dlgCopySheets

### PR DESCRIPTION
**Pull request by Lunalo**

### CODE REVIEW

**Comment by Lunalo** [comment]
Hi @dannyparsons  

I cant find the function for dlgCopySheets in the instat object.

I kindly request for your help here.

Thanks

**Comment by dannyparsons** [comment]
Ok thanks, I don't think the method exists yet so I'll create it soon.

File: instat/dlgCopySheet.vb [hunk 1]
```diff
@@ -1,3 +1,46 @@
...
+' along with this program.  If not, see <http://www.gnu.org/licenses/>.
+Imports instat.Translations
+Public Class dlgCopySheet
+    Public bFirstLoad As Boolean = True
+    Private Sub dlgCopySheet_Load(sender As Object, e As EventArgs) Handles MyBase.Load
+        autoTranslate(Me)
+        ucrBase.iHelpTopicID = 62
+        If bFirstLoad Then
+            SetDefaults()
+            InitialiseDialog()
```
Comment (dannyparsons) [inline instat/dlgCopySheet.vb]: Initialise should happen before SetDefaults

**Comment by dannyparsons** [comment]
In DeleteDataFrame, the event should be DataFrameChanged instead of Leave

**Comment by Lunalo** [comment]
@dannyparsons 

Please have a look at this

File: instat/dlgCopySheet.vb [hunk 2]
```diff
@@ -19,28 +19,36 @@ Public Class dlgCopySheet
...
-            InitialiseDialog()
             bFirstLoad = False
         Else
             ReopenDialog()
         End If
     End Sub
 
     Private Sub SetDefaults()
-
+        ucrNewSheetName.SetName("New_Name")
```
Comment (dannyparsons) [inline instat/dlgCopySheet.vb]: The default could be CurrentName & "_copy"

Comment (dannyparsons) [inline instat/dlgCopySheet.vb]: Then you could also do SetName when the data frame is changed as well so it keeps up to date.

**Comment by Lunalo** [comment]
@dannyparsons  

I have done that change in the above update

File: instat/dlgCopySheet.vb [hunk 3]
```diff
@@ -1,3 +1,54 @@
...
+            InitialiseDialog()
+            SetDefaults()
+            bFirstLoad = False
+        Else
+            ReopenDialog()
+        End If
+    End Sub
 
+    Private Sub SetDefaults()
+        ucrNewSheetName.SetName(ucrDataFrameCopySheets.cboAvailableDataFrames.SelectedItem & "_copy")
```
Comment (dannyparsons) [inline instat/dlgCopySheet.vb]: Can you also add this under the DataFrameChanged event so that it updates.

**Comment by Lunalo** [comment]
@dannyparsons 

This is now fixed

**Pull request merged**